### ISSUE TITLE
Create a wrapper for qio_decode_char_buf* to use String and Bytes module

### ISSUE THREAD

**Issue by e-kayrakli** [chapel-lang/chapel#15216]
We use `qio_decode_char_buf[_esc]` to decode byte buffers using UTF8 encoding, where the `_esc` variant allows us to use ["escaped sequences"](https://github.com/chapel-lang/chapel/pull/14743) in strings.  However; its an extern function, its use is cumbersome, and we repeat the [same thing](https://github.com/chapel-lang/chapel/blob/master/modules/internal/String.chpl#L1004-L1009) repeatedly in String.chpl and Bytes.chpl.

It'd be nice to create a helper in BytesStringCommon.chpl and use it in String.chpl and Bytes.chpl.

**Comment by priyank23** [comment]
@e-kayrakli I would like to work on this. I have analysed the String.chpl and Bytes.chpl . The above mentioned function is used significant number of times in String.chpl but not in Bytes.chpl. Can't we define a wrapper function in String.chpl only.

**Comment by parthsarthiprasad** [comment]
@e-kayrakli I was reading throught this as well , I'd like to work on it , I read through the code , Just a design related through , Since function signature is same for both functions , Can it be throught to make a single wrapper function , with another parameter to choose between the `qio_decode_char_buf_esc` or `qio_decode_char_buf` ? with something like
 ```chapel
qio_decode(ref chr:int(32),
            ref nbytes:c_int,
            buf:c_string,
            buflen:ssize_t, 
            esc: bool )
```

**Comment by arezaii** [comment]
This appears to be implemented and merged, can we close it?

**Comment by e-kayrakli** [comment]
Agreed. Resolved via #15292
